test/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDaoMockTest.java

```diff
@@ -0,0 +1,202 @@
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY;
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY_IDENTIFIER;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.sagebionetworks.bridge.dao.CriteriaDao;
+import org.sagebionetworks.bridge.dao.StudyConsentDao;
+import org.sagebionetworks.bridge.models.ClientInfo;
+import org.sagebionetworks.bridge.models.Criteria;
+import org.sagebionetworks.bridge.models.CriteriaContext;
+import org.sagebionetworks.bridge.models.subpopulations.Subpopulation;
+import org.sagebionetworks.bridge.models.subpopulations.SubpopulationGuid;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.PaginatedQueryList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+public class DynamoSubpopulationDaoMockTest {
+    
+    private static final Set<String> ALL_OF_GROUPS = Sets.newHashSet("a","b");
+    private static final Set<String> NONE_OF_GROUPS = Sets.newHashSet("c","d");
+    private static final SubpopulationGuid SUBPOP_GUID = SubpopulationGuid.create("AAA");
+    
+    private DynamoSubpopulationDao dao;
+    private DynamoDBMapper mapper;
+    private StudyConsentDao studyConsentDao;
+    private CriteriaDao criteriaDao;
+
+    @SuppressWarnings("unchecked")
+    @Before
+    public void before() {
+        dao = spy(new DynamoSubpopulationDao());
+        mapper = mock(DynamoDBMapper.class);
+        studyConsentDao = mock(StudyConsentDao.class);
+        criteriaDao = mock(CriteriaDao.class);
+        
+        dao.setMapper(mapper);
+        dao.setStudyConsentDao(studyConsentDao);
+        dao.setCriteriaDao(criteriaDao);
+        
+        // We need the copy constructor to work in order to verify CriteriaDao works.
+        when(criteriaDao.copyCriteria(any(), any())).thenAnswer(invocation -> {
+            String key = invocation.getArgumentAt(0, String.class);
+            Criteria criteria = invocation.getArgumentAt(1, DynamoCriteria.class);
+            Criteria actualCriteria = Criteria.create();
+            actualCriteria.setKey(key);
+            if (criteria != null) {
+                actualCriteria.setMinAppVersion(criteria.getMinAppVersion());
+                actualCriteria.setMaxAppVersion(criteria.getMaxAppVersion());
+                actualCriteria.setAllOfGroups(criteria.getAllOfGroups());
+                actualCriteria.setNoneOfGroups(criteria.getNoneOfGroups());
+            }
+            return actualCriteria;        
+        });
```

Comment:
Similarly, .thenCallRealMethod() or refactor copyCriteria() to somewhere where it can be shared.